graphql_compiler/macros/macro_edge/validation.py

```diff
@@ -61,6 +65,39 @@ def _validate_macro_ast_with_macro_directives(schema, ast, macro_directives):
                 .format(directive_definition.name, len(macro_data)))
 
 
+def _validate_macro_ast_directives(ast, inside_fold_scope=False):
+    """Check that the macro is using non-macro direcives properly."""
+    subselection_inside_fold_scope = inside_fold_scope
+    for directive in ast.directives:
+        name = directive.name.value
+        if name == FilterDirective.name:
+            pass
+        elif name == TagDirective.name:
+            pass
+        elif name == OutputDirective.name:
+            raise GraphQLInvalidMacroError(u'Macros should not have outputs.')
+        elif name == OutputSourceDirective.name:
+            raise GraphQLInvalidMacroError(u'Macros should not have output source directives.')
+        elif name == OptionalDirective.name:
+            pass
+        elif name == RecurseDirective.name:
+            pass
+        elif name == FoldDirective.name:
+            subselection_inside_fold_scope = True
+        elif name == MacroEdgeTargetDirective.name:
+            if inside_fold_scope:
+                raise GraphQLInvalidMacroError(
+                    u'The @macro_edge_target cannot be inside a fold scope.')
```

Comment:
add an `else` that raises `AssertionError` to make sure this matching is exhaustive

Reply:
Good catch. I had forgotten about `@macro_edge_definition`